go/teams/service_helper.go

```diff
@@ -144,6 +144,17 @@ func AddMember(ctx context.Context, g *libkb.GlobalContext, teamname, username s
 	if err != nil {
 		return keybase1.TeamAddMemberResult{}, err
 	}
+	existingUV, err := t.UserVersionByUID(ctx, uv.Uid)
+	if err == nil {
+		// Case where same UV (uid+seqno) already exists is covered by
+		// `t.IsMember` check above. This only checks if there is a reset
+		// member in the team to automatically remove them (so AddMember
+		// can function as a Re-Add).
+		if existingUV.EldestSeqno > uv.EldestSeqno {
+			return keybase1.TeamAddMemberResult{}, fmt.Errorf("EldestSeqno is going backwards")
```

Comment:
more descriptive error could be `"newer version of users already exists in team the team (%v > %v)", existingUV.EldestSeqno, uv.EldestSeqno`

Reply:
Thanks a lot!